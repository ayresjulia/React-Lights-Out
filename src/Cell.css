.Cell {
	height: 100px;
	width: 100px;
	cursor: pointer;

	border-radius: 50%;
}

.Cell-img {
	height: 100px;
	box-shadow: 0 3px 3px 0 rgba(0, 0, 0, 0.16), 0 3px 3px 0 rgba(0, 0, 0, 0.23);
	background-color: rgb(255, 255, 255, 0.8);
	border-radius: 50%;
}
.Cell-img-bw {
	height: 100px;
	filter: grayscale(100%);
}
