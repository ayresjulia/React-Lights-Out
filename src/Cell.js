import React from "react";
import "./Cell.css";

/** A single cell on the board.
 *
 * This has no state --- just two props:
 *
 * - flipCellsAroundMe: a function rec'd from the board which flips this
 *      cell and the cells around of it
 *
 * - isLit: boolean, is this cell lit?
 *
 * This handles clicks --- by calling flipCellsAroundMe
 *
 **/

function Cell ({ flipCellsAroundMe, isLit }) {
	const classes = `Cell ${isLit ? "Cell-lit" : ""}`;
	return (
		<td className={classes} onClick={flipCellsAroundMe} role="button">
			{isLit && (
				<img
					className="Cell-img"
					src="https://lh3.googleusercontent.com/proxy/RgoMpRZgtnNWhvmvnRjUhv57j_xBJLMV6iNSsneS5Gp0EzeVXmwbPlLxWnzto2aztG-3mB2KSqh57Ynfc1qrUphyQ2TR-d6l8q6lSbxq3QXGMqh_NA"
					alt="lightbulb"
				/>
			)}
			{!isLit && (
				<img
					className="Cell-img-bw"
					src="https://lh3.googleusercontent.com/proxy/RgoMpRZgtnNWhvmvnRjUhv57j_xBJLMV6iNSsneS5Gp0EzeVXmwbPlLxWnzto2aztG-3mB2KSqh57Ynfc1qrUphyQ2TR-d6l8q6lSbxq3QXGMqh_NA"
					alt="lightbulb"
				/>
			)}
		</td>
	);
}

export default Cell;
