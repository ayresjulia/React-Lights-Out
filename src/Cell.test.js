import { render } from "@testing-library/react";
import React from "react";
import Cell from "./Cell";

describe("smoke test", () => {
	it("renders without crashing", () => {
		render(<Cell />);
	});
});

describe("snapshot test", () => {
	it("matches snapshot", function () {
		const { asFragment } = render(<Cell />);
		expect(asFragment()).toMatchSnapshot();
	});
});
