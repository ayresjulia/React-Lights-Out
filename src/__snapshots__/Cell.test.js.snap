// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test matches snapshot 1`] = `
<DocumentFragment>
  <td
    class="Cell "
    role="button"
  >
    <img
      alt="lightbulb"
      class="Cell-img-bw"
      src="https://lh3.googleusercontent.com/proxy/RgoMpRZgtnNWhvmvnRjUhv57j_xBJLMV6iNSsneS5Gp0EzeVXmwbPlLxWnzto2aztG-3mB2KSqh57Ynfc1qrUphyQ2TR-d6l8q6lSbxq3QXGMqh_NA"
    />
  </td>
</DocumentFragment>
`;
